cocoa/src/toga_cocoa/widgets/selection.py

```diff
@@ -33,19 +32,49 @@ def rehint(self):
             max(self.interface._MIN_WIDTH, content_size.width)
         )
 
-    def remove_all_items(self):
+    def change_source(self, source):
         self.native.removeAllItems()
+        for item in source:
+            self.native.addItemWithTitle(self.interface._title_for_item(item))
+        self.interface.on_change(None)
 
-    def add_item(self, item):
-        self.native.addItemWithTitle(item)
+    def insert(self, index, item):
+        self.native.insertItemWithTitle(
+            self.interface._title_for_item(item),
+            atIndex=index,
+        )
 
-    def select_item(self, item):
-        self.native.selectItemWithTitle(item)
+        # If this is the first time item in the list, it will be automatically
+        # selected; trigger a change event.
+        if len(self.interface.items) == 1:
+            self.interface.on_change(None)
 
-    def get_selected_item(self):
-        selected = self.native.titleOfSelectedItem
-        if selected:
-            return str(selected)
+    def change(self, item):
+        index = self.interface._items.index(item)
```

Comment:
Imagine a list [A, B]. If B changes to become a second A, won't `index` return the first copy of A, and so the label of B will not be updated?

Reply:
No - index operates on instances, not value equality, so the reference is guaranteed to be to the second (old-B) instance.

Reply:
I think I understand the source of the confusion here - the docs for `index` and `find` aren't clear that index is an instance search, and find is an equality search. I'll clarify this.